spatialdata/_core/_spatialdata.py

```diff
@@ -0,0 +1,262 @@
+from __future__ import annotations
+
+from functools import singledispatch
+from types import MappingProxyType
+from typing import Any, Mapping, Optional, Union
+
+import numpy as np
+import zarr
+from anndata import AnnData
+from dask.array.core import Array as DaskArray
+from geopandas import GeoDataFrame
+from multiscale_spatial_image.multiscale_spatial_image import MultiscaleSpatialImage
+from ome_zarr.io import parse_url
+from spatial_image import SpatialImage
+
+from spatialdata._core.models import (
+    Image2DModel,
+    Image3DModel,
+    Label2DModel,
+    Label3DModel,
+    PointModel,
+    PolygonModel,
+    ShapeModel,
+    TableModel,
+)
+from spatialdata._io.write import (
+    write_image,
+    write_labels,
+    write_points,
+    write_polygons,
+    write_shapes,
+    write_table,
+)
+
+# schema for elements
+Label2d_s = Label2DModel()
+Label3D_s = Label3DModel()
+Image2D_s = Image2DModel()
+Image3D_s = Image3DModel()
+Polygon_s = PolygonModel
+Point_s = PointModel()
+Shape_s = ShapeModel()
+Table_s = TableModel()
+
+
+class SpatialData:
+    """Spatial data structure."""
```

Comment:
This is our main object, so I think we need a really complete docstring including the parameters for the `__init__` and the properties.

Reply:
Done.